clarity/data/HOA_tools_cec2.py

```diff
@@ -12,9 +12,38 @@
 logger = logging.getLogger(__name__)
 
 
-# Code for generation ambisonic rotation matrices
+def convert_a_to_b_format(
+    flu: np.ndarray, frd: np.ndarray, bld: np.ndarray, bru: np.ndarray
+):
+    """Converts 1st order A format audio into 1st order B format
+
+    Args:
+        flu (np.ndarray): Front-left-up audio
+        frd (np.ndarray): Front-right-down audio
+        bld (np.ndarray): Back-left-down audio
+        bru (np.ndarray): Back-right-up audio
 
+    Raises:
+        TypeError: input must be numpy array
+        ValueError: all inputs must have same dimensions
+
+    Returns:
+        nd.array: 4xN array containing B-format audio. indexed w,x,y,z
+    """
 
+    shapes = [flu.shape, frd.shape, bld.shape, bru.shape]
+    if not all(shape == shapes[0] for shape in shapes):
+        raise ValueError("All inputs need to have same dimensions")
+
+    w = flu + frd + bld + bru
```

Comment:
Does the different ways in which the arrays are added have any meaningful names that could be used?

`w` doesn't say anything about how they have been arranged for this instance it could be `sum_all` as they are all added. Not sure if there is anything more meaningful for this or the others but would be easier than refactoring at a subsequent time.